translator/navjoy_translator.py

```diff
@@ -0,0 +1,347 @@
+import argparse
+import copy
+import json
+import logging
+from collections import OrderedDict
+from datetime import datetime
+
+from translator.tools import date_tools, polygon_tools, wzdx_translator
+
+PROGRAM_NAME = 'NavJoy568Translator'
+PROGRAM_VERSION = '1.0'
+
+STRING_DIRECTION_MAP = {'north': 'northbound', 'south': 'southbound',
+                        'west': 'westbound', 'east': 'eastbound'}
+
+REVERSED_DIRECTION_MAP = {'northbound': 'southbound', 'southbound': 'northbound',
+                          'eastbound': 'westbound', 'westbound': 'eastbound'}
+
+
+def main():
+    inputfile, outputfile = parse_navjoy_arguments()
+    try:
+        navjoy_obj = json.loads(open(inputfile).read())
+    except ValueError as e:
+        raise ValueError(
+            'Invalid file type. Please specify a valid Json file!') from None
+    wzdx_obj = wzdx_creator(navjoy_obj)
+
+    location_schema = 'translator/sample files/validation_schema/wzdx_v3.1_feed.json'
+    wzdx_schema = json.loads(open(location_schema).read())
+
+    if not wzdx_translator.validate_wzdx(wzdx_obj, wzdx_schema):
+        logging.error(
+            'validation error more message are printed above. output file is not created because the message failed validation.')
+        return
+    with open(outputfile, 'w') as fwzdx:
+        fwzdx.write(json.dumps(wzdx_obj, indent=2))
+        print(
+            'Your wzdx message was successfully generated and is located here: ' + str(outputfile))
+
+
+# parse script command line arguments
+def parse_navjoy_arguments():
+    parser = argparse.ArgumentParser(
+        description='Translate iCone data to WZDx')
+    parser.add_argument('--version', action='version',
+                        version=f'{PROGRAM_NAME} {PROGRAM_VERSION}')
+    parser.add_argument('navjoyFile', help='navjoy file path')
+    parser.add_argument('--outputFile', required=False,
+                        default='navjoy_wzdx_translated_output_message.geojson', help='output file path')
+
+    args = parser.parse_args()
+    return args.navjoyFile, args.outputFile
+
+
+def wzdx_creator(messages, info=None, unsupported_message_callback=None):
+    if not messages:
+        return None
+   # verify info obj
+    if not info:
+        info = wzdx_translator.initialize_info(
+            '8d062f70-d53e-4029-b94e-b7fbcbde5885')
```

Comment:
Can we abstract this guid out to a class variable and change it from the cotrip one to differentiate between the two feeds on output?

Reply:
Updated to variable and created new GUID